tests/bmtestspolar/symfixedrankpolarfactory.m

```diff
@@ -0,0 +1,215 @@
+%       Title        = {Linear regression under fixed-rank constraints: a {R}iemannian approach},
+%       Author       = {Meyer, G. and Bonnabel, B. and Sepulchre, R.},
+%       Booktitle    = {{28th International Conference on Machine Learning}},
+%       Year         = {2011},
+%       Organization = {{ICML}}
+%     }
+% For second-order geometry, please cite the Manopt paper as well as the research paper:
+%     @Article{mishra2014fixedrank,
+%       Title   = {Fixed-rank matrix factorizations and {Riemannian} low-rank optimization},
+%       Author  = {Mishra, B. and Meyer, G. and Bonnabel, B. and Sepulchre, R.},
+%       Journal = {Computational Statistics},
+%       Year    = {2014},
+%       Number  = {3-4},
+%       Pages   = {591--621},
+%       Volume  = {29},
+%       Doi     = {10.1007/s00180-013-0464-z}
+%     }
+%
+%
+% See also symfixedrankYYfactory sympositivedefinitefactory
+    
+    % This file is part of Manopt: www.manopt.org.
+    % Original author: Bamdev Mishra, April 05, 2019.
+    % Contributors:
+    % Change log:
+    %
+    
+    M.name = @() sprintf('UBU'' quotient manifold of %dx%d matrices of rank %d', m, m, k);
+    
+    M.dim = @() m*k - k*(k-1)/2;
+    
+    % Choice of the metric on the orthnormal space is motivated by the symmetry present in the
+    % space. The metric on the positive definite space is its natural metric.
+    M.inner = @(X, eta, zeta) eta.U(:).'*zeta.U(:)  ...
+        + trace( (X.B\eta.B) * (X.B\zeta.B) );
+    
+    M.norm = @(X, eta) sqrt(M.inner(X, eta, eta));
+    
+    M.dist = @(x, y) error('symfixedrankpolarfactory.dist not implemented yet.');
+    
+    M.typicaldist = @() 10*k;
+    
+    skew = @(X) .5*(X-X');
+    symm = @(X) .5*(X+X');
+    stiefel_proj = @(U, H) H - U*symm(U'*H);
+    
+    M.egrad2rgrad = @egrad2rgrad;
+    function rgrad = egrad2rgrad(X, egrad)
+        rgrad.U = stiefel_proj(X.U, egrad.U);
+        rgrad.B = X.B*symm(egrad.B)*X.B;
+    end
+    
+    
+    M.ehess2rhess = @ehess2rhess;
+    function Hess = ehess2rhess(X, egrad, ehess, eta)
+        
+        % Riemannian gradient for the factor B.
+        rgrad.B = X.B*symm(egrad.B)*X.B;
+        
+        % Directional derivatives of the Riemannian gradient.
+        Hess.U = ehess.U - eta.U*symm(X.U'*egrad.U);
+        Hess.U = stiefel_proj(X.U, Hess.U);
+        
+        Hess.B = X.B*symm(ehess.B)*X.B +  2*symm(eta.B*symm(egrad.B)*X.B);
+        
+        % Correction factor for the non-constant metric on the factor B.
+        Hess.B = Hess.B - symm(eta.B*(X.B\rgrad.B));
+        
+        % Projection onto the horizontal space.
+        Hess = M.proj(X, Hess);
+    end
+    
+    
+    M.proj = @projection;
+    function etaproj = projection(X, eta)
+        % First, projection onto the tangent space of the total space.
+        eta.U = stiefel_proj(X.U, eta.U);
+        eta.B = symm(eta.B);
+        
+        % Then, projection onto the horizontal space.
+        SS = X.B*X.B;
```

Comment:
```suggestion
        SS = X.B*X.B';
```

I know it's technically the same since X.B is symmetric, but this form (I believe?) lets Matlab know that the result has to be symmetric, and it will enforce that.